node/src/main/scala/com/wavesplatform/database/package.scala

```diff
@@ -645,11 +637,11 @@ package object database {
   def writeTransaction(v: (TxMeta, Transaction)): Array[Byte] = {
     val (m, tx) = v
     val ptx = tx match {
-      case lps: PBSince if !lps.isProtobufVersion => TD.LegacyBytes(ByteString.copyFrom(tx.bytes()))
-      case _: GenesisTransaction                  => TD.LegacyBytes(ByteString.copyFrom(tx.bytes()))
-      case _: PaymentTransaction                  => TD.LegacyBytes(ByteString.copyFrom(tx.bytes()))
-      case et: EthereumTransaction                => TD.EthereumTransaction(ByteString.copyFrom(et.bytes()))
-      case _                                      => TD.WavesTransaction(PBTransactions.protobuf(tx))
+      case lps: PBSince with VersionedTransaction if !PBSince.affects(lps) => TD.LegacyBytes(ByteString.copyFrom(tx.bytes()))
+      case _: GenesisTransaction                                           => TD.LegacyBytes(ByteString.copyFrom(tx.bytes()))
+      case _: PaymentTransaction                                           => TD.LegacyBytes(ByteString.copyFrom(tx.bytes()))
+      case et: EthereumTransaction                                         => TD.EthereumTransaction(ByteString.copyFrom(et.bytes()))
+      case _                                                               => TD.WavesTransaction(PBTransactions.protobuf(tx))
```

Comment:
Wondering, why not:
```suggestion
      case lps: PBSince with VersionedTransaction if PBSince.affects(lps) => TD.WavesTransaction(PBTransactions.protobuf(tx))
      case et: EthereumTransaction                                        => TD.EthereumTransaction(ByteString.copyFrom(et.bytes()))
      case _                                                              => TD.LegacyBytes(ByteString.copyFrom(tx.bytes()))
```
It looks simpler

Reply:
Done